src/components/MessageActions/MessageActions.new.tsx

```diff
@@ -0,0 +1,367 @@
+/* eslint-disable sort-keys */
+import clsx from 'clsx';
+import React, { ComponentPropsWithoutRef, PropsWithChildren, useMemo, useState } from 'react';
+
+import { ReactionSelectorWithButton } from '../Reactions/ReactionSelectorWithButton';
+import {
+  useChannelActionContext,
+  useChatContext,
+  useMessageContext,
+  useTranslationContext,
+} from '../../context';
+import { ActionsIcon, ReactionIcon as DefaultReactionIcon, ThreadIcon } from '../Message/icons';
+import { ACTIONS_NOT_WORKING_IN_THREAD, isUserMuted } from '../Message/utils';
+import { useUserRole } from '../Message/hooks';
+import { DialogAnchor, useDialog, useDialogIsOpen } from '../Dialog';
+import { MessageActionsWrapper } from './MessageActions';
+
+type MessageAction = {
+  Component: React.ComponentType;
+  placement: 'quick' | 'dropdown';
+  type:
+    | 'reply'
+    | 'react'
+    | 'block'
+    | 'delete'
+    | 'edit'
+    | 'mute'
+    | 'flag'
+    | 'pin'
+    | 'quote'
+    | 'markAsUnread'
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    | (string & {});
+};
+
+const DefaultMessageActions_UNSTABLE = {
+  dropdown: {
+    // eslint-disable-next-line react/display-name
+    Quote: () => {
```

Comment:
`displayName` is quite a useful thing in dev tools. Instead of disabling the eslint rule, why don't we use:
```ts
Quote: function Quote() { ... }
```